examples/consistency_models/train_consistency_distillation.py

```diff
@@ -0,0 +1,701 @@
+            path = os.path.basename(args.resume_from_checkpoint)
+        else:
+            # Get the most recent checkpoint
+            dirs = os.listdir(args.output_dir)
+            dirs = [d for d in dirs if d.startswith("checkpoint")]
+            dirs = sorted(dirs, key=lambda x: int(x.split("-")[1]))
+            path = dirs[-1] if len(dirs) > 0 else None
+
+        if path is None:
+            accelerator.print(
+                f"Checkpoint '{args.resume_from_checkpoint}' does not exist. Starting a new training run."
+            )
+            args.resume_from_checkpoint = None
+        else:
+            accelerator.print(f"Resuming from checkpoint {path}")
+            accelerator.load_state(os.path.join(args.output_dir, path))
+            global_step = int(path.split("-")[1])
+
+            initial_global_step = global_step * args.gradient_accumulation_steps
+            first_epoch = global_step // num_update_steps_per_epoch
+    else:
+        initial_global_step = 0
+
+    progress_bar = tqdm(
+        range(0, max_train_steps),
+        initial=initial_global_step,
+        desc="Steps",
+        # Only show the progress bar once on each machine.
+        disable=not accelerator.is_local_main_process,
+    )
+
+
+
+    # Train!
+    for epoch in range(first_epoch, args.num_epochs):
+        model.train()
+        for step, batch in enumerate(train_dataloader):
+            clean_images = batch["input"]
+            labels = batch["labels"]
+            # Sample noise that we'll add to the images
+            noise = torch.randn(clean_images.shape).to(clean_images.device)
+            # Sample a random timestep for each image, TODO - allow different timesteps in a batch
+            index = torch.randint(
+                0, noise_scheduler.config.num_train_timesteps-1,  (1,), device=clean_images.device
+            ).long()
+            timestep = timesteps[index]
+            timestep_prev = timestep + 1
+            # TODO, we should have an add noise in the scheduler maybe?
+            noised_image = clean_images + noise*append_dims(timestep, clean_images.ndim)
+            scaled_timesteps = noise_scheduler.scale_timestep(timestep)
+            scaled_timesteps_prev = noise_scheduler.scale_timestep(timestep_prev) 
+            target_model_ema.copy_to(target_model.parameters())
+
+            with accelerator.accumulate(model):
+                # Predict the noise residual
+                model_output = model(noised_image, scaled_timesteps, class_labels=labels).sample
+                distiller = noise_scheduler.step(
+                    model_output, timestep, noised_image, use_noise=False
+                ).prev_sample
+
+                # Heun Solver to get previous timestep image using teacher model
+                # TODO - make this cleaner
+                samples = noised_image
+                x = samples
+                model_output = teacher_model(x, scaled_timesteps, class_labels=labels).sample
```

Comment:
I think the teacher noised image `x` here should also be scaled appropriately with respect to how the teacher model (for example, `google/ddpm-cifar10-32`) is trained. When we get the teacher model

https://github.com/dg845/diffusers/blob/cd460ca2f5107ca6053e27c6e92f0aee2248627c/examples/consistency_models/train_consistency_distillation.py#L414

we should also be getting the teacher scheduler (e.g. `teacher_scheduler = DiffusionPipeline.from_pretrained(<teacher checkpoint>).scheduler`) and then calling its `scale_model_input` method, which is analogous to how the original implementation [gets the teacher diffusion with the teacher model](https://github.com/openai/consistency_models/blob/main/scripts/cm_train.py#L80-L82) and then [calls `teacher_diffusion`'s `denoise` method](https://github.com/openai/consistency_models/blob/main/cm/karras_diffusion.py#L136-L140) to appropriately scale the teacher input.

Similarly, the teacher timestep should be scaled according to the teacher pipeline's scheduler, which is in general different from `scaled_timesteps`, which is scaled according to the student consistency model's scheduler.

Reply:
For scaling, I think yeah it might make sense to use the scheduler of the teacher, need to investigate this more